src/pg_cron.c

```diff
@@ -1266,6 +1629,509 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
+									 task->jobId, cmdStatus, cmdTuples)));
+			}
+
+			break;
+		}
+
+		case PGRES_BAD_RESPONSE:
+		case PGRES_FATAL_ERROR:
+		{
+			task->errorMessage = strdup(PQresultErrorMessage(result));
+			task->freeErrorMessage = true;
+			task->pollingStatus = 0;
+			task->state = CRON_TASK_ERROR;
+
+			if (CronLogRun)
+				UpdateJobRunDetail(task->runId, NULL, GetCronStatus(CRON_STATUS_FAILED), task->errorMessage, NULL, &end_time);
+
+			PQclear(result);
+
+			return;
+		}
+
+		case PGRES_COPY_IN:
+		case PGRES_COPY_OUT:
+		case PGRES_COPY_BOTH:
+		{
+			/* cannot handle COPY input/output */
+			task->errorMessage = "COPY not supported";
+			task->pollingStatus = 0;
+			task->state = CRON_TASK_ERROR;
+
+			if (CronLogRun)
+				UpdateJobRunDetail(task->runId, NULL, GetCronStatus(CRON_STATUS_FAILED), task->errorMessage, NULL, &end_time);
+
+			PQclear(result);
+
+			return;
+		}
+
+		case PGRES_TUPLES_OK:
+		case PGRES_EMPTY_QUERY:
+		case PGRES_SINGLE_TUPLE:
+		case PGRES_NONFATAL_ERROR:
+		default:
+		{
+			int tupleCount = PQntuples(result);
+			char *rowString = ngettext("row", "rows",
+										   tupleCount);
+			char  rows[MAXINT8LEN + 1];
+			char  outputrows[MAXINT8LEN + 4 + 1];
+
+			pg_lltoa(tupleCount, rows);
+			snprintf(outputrows, sizeof(outputrows), "%s %s", rows, rowString);
+
+			if (CronLogRun)
+				UpdateJobRunDetail(task->runId, NULL, GetCronStatus(CRON_STATUS_SUCCEEDED), outputrows, NULL, &end_time);
+
+			if (CronLogStatement)
+			{
+				ereport(LOG, (errmsg("cron job " INT64_FORMAT " completed: "
+									 "%d %s",
+									 task->jobId, tupleCount,
+									 rowString)));
+			}
+
+			break;
+		}
+
+	}
+
+	PQclear(result);
+}
+
+static void
+GetBgwTaskFeedback(shm_mq_handle *responseq, CronTask *task)
+{
+
+	TimestampTz end_time;
+
+	Size            nbytes;
+	void       *data;
+	char            msgtype;
+	StringInfoData  msg;
+	shm_mq_result res;
+
+	end_time = GetCurrentTimestamp();
+	/*
+	 * Message-parsing routines operate on a null-terminated StringInfo,
+	 * so we must construct one.
+	 */
+	for (;;)
+	{
+		/* Get next message. */
+		res = shm_mq_receive(responseq, &nbytes, &data, false);
+
+		if (res != SHM_MQ_SUCCESS)
+			break;
+		initStringInfo(&msg);
+		resetStringInfo(&msg);
+		enlargeStringInfo(&msg, nbytes);
+		msg.len = nbytes;
+		memcpy(msg.data, data, nbytes);
+		msg.data[nbytes] = '\0';
+		msgtype = pq_getmsgbyte(&msg);
+		switch (msgtype)
+		{
+			case 'N':
+			case 'E':
+				{
+					ErrorData	edata;
+					StringInfoData  display_msg;
+
+					pq_parse_errornotice(&msg, &edata);
+					initStringInfo(&display_msg);
+					bgw_generate_returned_message(&display_msg, edata);
+
+					if (CronLogRun)
+					{
+
+						if (edata.elevel >= ERROR)
+							UpdateJobRunDetail(task->runId, NULL, GetCronStatus(CRON_STATUS_FAILED), display_msg.data, NULL, &end_time);
+						else
+							UpdateJobRunDetail(task->runId, NULL, GetCronStatus(CRON_STATUS_SUCCEEDED), display_msg.data, NULL, &end_time);
```

Comment:
It's kind of neat to see the notices appear in the table. They do get overwritten once the task is done.   Is that intentional? (I don't mind this behaviour)